npep/npep-133.md

```diff
@@ -0,0 +1,89 @@
+# NPEP-133: FQDN Selector for Egress Traffic
+
+* Issue:
+  [#133](https://github.com/kubernetes-sigs/network-policy-api/issues/133)
+* Status: Provisional
+
+## TLDR
+
+This enhancement proposes adding a new selector to specify egress peers using
+Fully Qualified Domain Names (FQDNs). 
+
+## Goals
+
+* Provide a selector to specify egress peers using a Fully Qualified Domain Name
+  (for example `kubernetes.io`).
+* Support a restricted set of regex matching capabilities when specifying FQDNs. 
+
+## Non-Goals
+
+* This enhancement does not include a FQDN selector for allowing ingress
+  traffic.
+* This enhancement does not include any L7 matching or filtering capabilities,
+  like matching HTTP traffic or URL paths.
+    * This selector should not control what DNS records are resolvable from a
+      particular workload.
+* This enhancement does not provide a mechanism for selecting in-cluster
+  endpoints using FQDNs. This is explicitly disallowed by the spec.
+    * To select Pods, Nodes, API Server, AdminNetworkPolicy has more first party
+      selector with better UX.
+* This enhancement does not specify the details of how traffic is routed to the
+  specified destination. For example, it does not prescribe details around NAT
+  or egress gateways.
+* This enhancement does not require a particular mechanism for securing DNS
+  resolution (e.g. DNSSEC or DNS-over-TLS).
+    * TODO: Is this up to implementations? Maybe an optional enum describing
+      what security mechanims to use and providers can chose whther to support
```

Comment:
It doesn't require "particular" mechanisms for securing DNS? Does it require _any_ mechanisms?

Reply:
clarified.